homeassistant/helpers/triggers/webhook.py

```diff
@@ -39,7 +37,7 @@ async def async_attach_trigger(hass, config, action, automation_info):
     """Trigger based on incoming webhooks."""
     webhook_id = config.get(CONF_WEBHOOK_ID)
     hass.components.webhook.async_register(
-        AUTOMATION_DOMAIN,
+        "automation",
```

Comment:
We should source this value from `automation_info`, since other integrations can now start using triggers.

Reply:
I'm glad you said something about this because if you hadn't I was going to ask. This was the only thing I really didn't know how to handler because I didn't know how this value was used.

Now this (like other triggers) will be used in an automation's trigger section, or in an automation's action or script's sequence via a wait_for_trigger action. What key should I use? From an automation's trigger it's clear to use the value "automation." But what about wait_for_trigger, which can be used anywhere the script helper (aka script syntax) is used. What value do I use for that?

Reply:
We might want to consider initializing the helper Script objects with an "Owner" domain that we pass down.

Reply:
I think that's a good idea.

Also, I want to make the name parameter to the `Script` constructor required as well. I found a bug in the functions I added for stopping script runs at shutdown. If a name was not provided it causes an exception. (This isn't an issue for automations and scripts, because they both pass in name, but there are 16 other users of the script helper, and many of those do _not_ pass in a name.) Even if I fixed that bug to deal with name being `None`, it's still not very useful logging a message that the run is being stopped if it can't say which one. Lastly, I think it would be good to add yet another (optional) parameter, namely description, and use it in the INFO log message that currently says "Running script". That way it could say "Running automation actions", or "Running script", or "Running on script", etc. I've already gotten feedback like, "Why does it say running script when it's actually an automation?"

So, how about I do these changes in a separate PR, and then I can use it here (i.e., use the owner domain)?

Reply:
@balloob what domain should be used? E.g., it's clear for the automation & script integrations, but what about the other 16 users? Many use `Script` in their `media_player`, or `light`, or `switch` etc. platform code. So use that? I mean `media_player`, `light`, etc.? Or, the integration name, like `philips_js`, or `knx`, or `template` (which uses it in template fan, lock, alarm_control_panel, light, cover, vacuum & switch)?

Reply:
It should be attributed to the owner, so look at the integration folder in which it is located. So in the case of `philips_js.media_player` it will be `philips_js`.

Reply:
#38763